src/main/scala/com/fulcrumgenomics/bam/ClipBam.scala

```diff
@@ -112,36 +121,160 @@ class ClipBam
-    this.clipper.clip5PrimeEndOfRead(r2, readTwoFivePrime)
-    this.clipper.clip3PrimeEndOfRead(r2, readTwoThreePrime)
-
-    if (clipOverlappingReads && r1.mapped && r2.mapped && r1.refIndex == r2.refIndex && r1.pairOrientation == PairOrientation.FR) {
-      val (f,r) = if (r1.negativeStrand) (r2, r1) else (r1, r2)
-      clipOverlappingReads(f, r)
+    val numReadOneFivePrime  = this.clipper.clip5PrimeEndOfRead(r1, readOneFivePrime)
+    val numReadOneThreePrime = this.clipper.clip3PrimeEndOfRead(r1, readOneThreePrime)
+    val numReadTwoFivePrime  = this.clipper.clip5PrimeEndOfRead(r2, readTwoFivePrime)
+    val numReadTwoThreePrime = this.clipper.clip3PrimeEndOfRead(r2, readTwoThreePrime)
+
+    val (numOverlappingBasesReadOne, numOverlappingBasesReadTwo) = {
+      if (clipOverlappingReads && r1.mapped && r2.mapped && r1.refIndex == r2.refIndex && r1.pairOrientation == PairOrientation.FR) {
+        val (f,r) = if (r1.negativeStrand) (r2, r1) else (r1, r2)
+        clipOverlappingReads(f, r)
+      } else (0, 0)
+    }
+
+    r1Metric.foreach { m =>
+      m.update(
+        rec                 = r1,
+        priorBasesClipped   = priorBasesClippedReadOne,
+        numFivePrime        = numReadOneFivePrime,
+        numThreePrime       = numReadOneThreePrime,
+        numOverlappingBases = numOverlappingBasesReadOne
+      )
+    }
+
+    r2Metric.foreach { m =>
+      m.update(
+        rec                 = r2,
+        priorBasesClipped   = priorBasesClippedReadTwo,
+        numFivePrime        = numReadTwoFivePrime,
+        numThreePrime       = numReadTwoThreePrime,
+        numOverlappingBases = numOverlappingBasesReadTwo
+      )
     }
   }
 
   /** Clips overlapping reads, where both ends of the read pair are mapped to the same chromosome, and in FR orientation. */
-  protected def clipOverlappingReads(f: SamRecord, r: SamRecord): Unit = {
+  protected def clipOverlappingReads(f: SamRecord, r: SamRecord): (Int, Int) = {
+    var numOverlappingBasesReadOne: Int = 0
+    var numOverlappingBasesReadTwo: Int = 0
     // What we really want is to trim by the number of _reference_ bases not read bases,
     // in order to eliminate overlap.  We could do something very complicated here, or
     // we could just trim read bases in a loop until the overlap is eliminated!
     while (f.end >= r.start && f.mapped && r.mapped) {
       val lengthToClip = f.end - r.start + 1
       val firstHalf    = lengthToClip / 2
       val secondHalf   = lengthToClip - firstHalf // safe guard against rounding on odd lengths
-      this.clipper.clip3PrimeEndOfAlignment(f, firstHalf)
-      this.clipper.clip3PrimeEndOfAlignment(r, secondHalf)
+      numOverlappingBasesReadOne += this.clipper.clip3PrimeEndOfAlignment(f, firstHalf)
+      numOverlappingBasesReadTwo += this.clipper.clip3PrimeEndOfAlignment(r, secondHalf)
+    }
+    (numOverlappingBasesReadOne, numOverlappingBasesReadTwo)
+  }
+}
+
+sealed trait ReadType extends EnumEntry
```

Comment:
Not that we need to do this now, but I feel like we've done similar things elsewhere so the next time we need to output things by R1/R2/Frag we should try and remember this is here and elevate it.